ci/azure-pipelines-nightly.yml

```diff
@@ -108,8 +108,8 @@ jobs:
     displayName: "Run all ROM E2E tests"
   - template: ../ci/publish-bazel-test-results.yml
 
-- job: nightly_tagged_sw_tests
-  displayName: "Software tests explicitly tagged as 'nightly'"
+- job: nightly_tagged_tests
```

Comment:
I think this second commit has two parts:
- Restrict the nightly pipeline less, so that it's not just running software tests
- Avoid running nightly tests in `run-fpga-tests.sh`

Do those changes need to be in the same commit? If not, I'd be tempted to split the commit in two.